### ISSUE TITLE
vis_heap_chunks : TypeError: '<=' not supported between instances of 'int' and 'NoneType'

### ISSUE THREAD

**Issue by yooo1999** [pwndbg/pwndbg#1550]
I run the last commit ( 0f67d08acb2afec32549e5bf60e222efcab6428d )
When I press `vis` in the main thread everything good,
but if I move to another thread
`thread 2`
and then press `vis` I got an exception :
```
vis_heap_chunks: An unknown error occurred when running this command.
Maybe you can try to determine the libc symbols addresses manually, set them appropriately and re-run this command. For this, see the `heap_config` command output and set the `main_arena`, `mp_`, `global_max_fast`, `tcache` and `thread_arena` addresses.
'vis_heap_chunks': Visualize chunks on a heap.


Default to the current arena's active heap.
Traceback (most recent call last):
  File "/home/USER/pwndbg/pwndbg/commands/__init__.py", line 184, in __call__
    return self.function(*args, **kwargs)
  File "/home/USER/pwndbg/pwndbg/commands/__init__.py", line 315, in _OnlyWhenRunning
    return function(*a, **kw)
  File "/home/USER/pwndbg/pwndbg/commands/__init__.py", line 395, in _OnlyWithResolvedHeapSyms
    return _try2run_heap_command(function, a, kw)
  File "/home/USER/pwndbg/pwndbg/commands/__init__.py", line 377, in _try2run_heap_command
    raise err
  File "/home/USER/pwndbg/pwndbg/commands/__init__.py", line 358, in _try2run_heap_command
    return function(*a, **kw)
  File "/home/USER/pwndbg/pwndbg/commands/__init__.py", line 340, in _OnlyWhenHeapIsInitialized
    return function(*a, **kw)
  File "/home/USER/pwndbg/pwndbg/commands/heap.py", line 673, in vis_heap_chunks
    heap_region = arena.active_heap
  File "/home/USER/pwndbg/pwndbg/heap/ptmalloc.py", line 591, in active_heap
    self._active_heap = Heap(self.top, arena=self)
  File "/home/USER/pwndbg/pwndbg/heap/ptmalloc.py", line 355, in __init__
    if addr in sbrk_region:
  File "/home/USER/pwndbg/pwndbg/lib/memory.py", line 146, in __contains__
    return self.start <= addr < self.end
TypeError: '<=' not supported between instances of 'int' and 'NoneType'
```

**Comment by CptGibbon** [comment]
I get the sneaking suspicion that this happens when attempting to use `vis_heap_chunks` when stopped in a thread which isn't attached to an arena. Can you confirm please?

**Comment by yooo1999** [comment]
@CptGibbon How can I check if thread doesn't attach any arena ?
Or how can I print for each arena all the thread ID  that belong to this arena ?

**Comment by CptGibbon** [comment]
For testing purposes, just create a new thread and stop it before it makes any allocations, just to be sure 👍

**Comment by yooo1999** [comment]
I sorry, I'm not pretty sure how to simulate that ...

**Comment by disconnect3d** [comment]
> I get the sneaking suspicion that this happens when attempting to use `vis_heap_chunks` when stopped in a thread which isn't attached to an arena. Can you confirm please?

I think this is true? I can reproduce this bug with:

```c
#include <pthread.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

void *thread_function(void *arg) {
    usleep(1000);
    exit(1);
    return NULL;
}

void *useless_thread(void *arg) {
    while (1) {
        usleep(1000);
    }
    return NULL;
}

void break_here() {}

int main() {
    pthread_t thread;
    int res = pthread_create(&thread, NULL, thread_function, NULL);
    if (res != 0) {
        perror("Thread creation failed");
        exit(EXIT_FAILURE);
    }
    int res2 = pthread_create(&thread, NULL, useless_thread, NULL);
    if (res2 != 0) {
        perror("Thread creation failed");
        exit(EXIT_FAILURE);
    }

    break_here();
    usleep(4000);

    return 0;
}
```

Compile as `gcc threads.c -lpthread -pthread`. 

The problem is that the `self.top` is `None`.

**Comment by disconnect3d** [comment]
cc: @lebr0nli

**Comment by lebr0nli** [comment]
> The problem is that the `self.top` is `None`.

I can reproduce it, too.
This problem is mainly cause by this:
https://github.com/pwndbg/pwndbg/blob/ee832c80d281c903efdba0306d090dcacd267518/pwndbg/heap/ptmalloc.py#L1128
If the current thread doesn't allocate arena, `thread_arena` points to `NULL`, but we didn't check it.
So `int(self._gdbValue["top"])` cause `gdb.MemoryError` and cause `self.top` is `None`.
I'll like to do the fix for this issue at [#1579](https://github.com/pwndbg/pwndbg/pull/1579) since there's a [related commit](https://github.com/pwndbg/pwndbg/pull/1579/commits/f42354613b805e5735b2cca305d5198f25a484a3) in that PR.
